balena/models/device_os.py

```diff
@@ -40,8 +40,8 @@ def __ne__(self, other):
     return K
 
 def sort_version(x, y):
-    if semver.VersionInfo.isvalid(x['raw_version']) and semver.VersionInfo.isvalid(y['raw_version']):
-        return semver.compare(x['raw_version'], y['raw_version'])
+    if semver.VersionInfo.isvalid(x['stripped_version']) and semver.VersionInfo.isvalid(y['stripped_version']):
+        return semver.compare(x['stripped_version'], y['stripped_version'])
```

Comment:
Is that just to avoid the `1.2.3.prod` buggy case?
PS: Reminder that the node balena-semver also takes revisions into consideration while sorting, while the upstream `semver` packages (node & python) simply ignore all build metadata parts in `.compare`.

Reply:
I've implemented a `compare` function based on `balena-semver.compare` so all build metadata parts will not be ignored.